processor/metricstransformprocessor/metrics_transform_processor.go

```diff
@@ -406,3 +412,45 @@ func (mtp *metricsTransformProcessor) compareTimestamps(t1 *timestamppb.Timestam
 
 	return t1.Seconds < t2.Seconds || (t1.Seconds == t2.Seconds && t1.Nanos < t2.Nanos)
 }
+
+func labelMatched(filterLabels map[string]string, metric *metricspb.Metric, isRegexp bool) bool {
+	if len(filterLabels) == 0 {
+		return true
+	}
+
+	for key, value := range filterLabels {
+		keyFound := false
+
+		for idx, label := range metric.MetricDescriptor.LabelKeys {
+			if label.Key != key {
+				continue
+			}
+
+			keyFound = true
+			for _, timeseries := range metric.Timeseries {
+				if isRegexp {
+					re := regexp.MustCompile(value)
+					if !re.MatchString(timeseries.LabelValues[idx].Value) {
+						return false
+					}
+				} else {
+					if timeseries.LabelValues[idx].Value != value {
+						return false
+					}
+				}
+				break
+			}
+		}
+
+		// if a label-key is not found and the label-value is non-empty, return false
+		if !keyFound && !(value == "" || isEmptyExp(value)) {
```

Comment:
I don't think this logic matches the description in the comment.  It looks like a non-empty label value that is matched by a regexp that could also match an empty label value will hit this condition.  What is intended to be tested here?

Reply:
We have two different use case here. Say, in our config file we have the following labels to match,
1. `{"container": "my-container"}` where the value of the key `container` is non-empty.
2. `{"container": ""}` where the value of the key `container` is empty-string. 

For first case, if we don't find a key, we return false. But for the second case, if we don't find the key, we need to return true which is expected. 

As we don't have a way to make sure a given key is not present, I wrote this logic as alternative. If we want to confirm that a given label is not present, we set the value as empty-string. 

Hope it helps to explain.

Reply:
Updated the comment to make it more clear.